config/githooks/README.md

```diff
@@ -0,0 +1,44 @@
+                    Finite Element Discretization Library
+                                   __
+                       _ __ ___   / _|  ___  _ __ ___
+                      | '_ ` _ \ | |_  / _ \| '_ ` _ \
+                      | | | | | ||  _||  __/| | | | | |
+                      |_| |_| |_||_|   \___||_| |_| |_|
+
+                               https://mfem.org
+
+
+This directory contains recommended git hooks, which are scripts that can be
+used to improve your development experience with MFEM:
+
+### The hooks
+
+* `pre-commit` is a hook that will be applied before each commit and run
+`astyle` on the code. This will ensure that your changes comply with the MFEM
+code styling guidelines.
+
+* `pre-push` is a hook that will be applied before each push to run a quick set
+of tests that verify that your files headers are in compliance, and that you did
+not add any large files to the repo.
+
+### Setup
+
+To setup a git hook, create a symlink to it from the `.git/hooks` directory.
```

Comment:
Should we maybe add `make hooks` target to make this even simpler?

Reply:
Tempting, but it would be forcing to adopt all the hooks. I think it's good to know how to set the hooks, so one knows how to unset them ;).

Reply:
I am on the fence myself.

On one hand I totally agree with you. On another, the easier it is to use the hooks, the more people will use them.

@pazner and @Jacobfaib, what do you think?

Reply:
I think `make hooks` is a good idea. The README could include something like 
```
To setup the git hooks, run `make hooks`, which creates symlinks to the hooks in
the `.git/hooks` directory. Individual hooks can be enabled by manually creating
symlinks.
```

Reply:
I like @pazner copy/paste-ready suggestions.